activiti-core/activiti-api-impl/activiti-api-process-model-impl/src/main/java/org/activiti/api/runtime/model/impl/ProcessVariablesMapDeserializer.java

```diff
@@ -45,13 +46,31 @@ public ProcessVariablesMapDeserializer(ConversionService conversionService) {
             JsonNode entryValue = entry.getValue();
 
             if(!entryValue.isNull()) {
-                String type = entryValue.get("type").textValue();
-                String value = entryValue.get("value").asText();
+                if (entryValue.get("type") != null && entryValue.get("value") != null) {
+                    String type = entryValue.get("type").textValue();
+                    String value = entryValue.get("value").asText();
 
-                Class<?> clazz = ProcessVariablesMapTypeRegistry.forType(type);
-                Object result = conversionService.convert(value, clazz);
+                    Class<?> clazz = ProcessVariablesMapTypeRegistry.forType(type);
+                    Object result = conversionService.convert(value, clazz);
+
+                    if(ObjectValue.class.isInstance(result)) {
+                        result = ObjectValue.class.cast(result)
+                                                  .getObject();
+                    }
+
+                    map.put(name, result);
+                }
+                else {
+                    Object value = null;
+                    try {
+                        value = new ObjectMapper().treeToValue(entryValue,
+                                                               Object.class);
+                    } catch (JsonProcessingException e) {
+                        e.printStackTrace();
```

Comment:
Please, use logger instead of `printStackTrace()`.

Reply:
done.